tools/image-builder/templates/packer_user_data.tpl

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/bash
+
+###################################################
+# Copyright (C) IBM Corp. 2023 All Rights Reserved.
+# Licensed under the Apache License v2.0
+###################################################
+
+# Install required packages
+
+export HOME=/root  # Setting this as a path because got to see error on user data scripts, we can revisit this logic later
+
+sudo yum install -y wget unzip jq
+
+# Decode and extract the base64-encoded content
+
+sudo mkdir -p ${target_dir}
+echo "Decoding and saving base64-encoded content [necessary to retrieve the original compressed file]"
+echo "${encoded_compute}" | base64 -d > ${target_dir}/compressed_compute.tar.gz
+
+# Extract the tar.gz file
+echo "Unpacking the contents of the compressed file"
+tar -xzf ${target_dir}/compressed_compute.tar.gz -C ${target_dir}
+rm -f ${target_dir}/compressed_compute.tar.gz
+
+echo "Packer installation started"
+
+# Download and unzip Packer
+wget https://releases.hashicorp.com/packer/1.11.1/packer_1.11.1_linux_amd64.zip
+unzip packer_1.11.1_linux_amd64.zip
+
+# Move Packer to /usr/local/bin
+sudo mv packer /usr/local/bin/
+
+# Create a symlink to /usr/sbin
+sudo ln -sf /usr/local/bin/packer /usr/sbin/packer
+
+install_with_retry() {
+    local cmd="$1"
+    local retries="$2"
+    local count=0
+
+    until $cmd || [ $count -eq "$retries" ]; do
+        echo "Installation failed. Retrying..."
+        sleep 5  # Adjust sleep duration between retries as needed
+        count=$((count + 1))
+    done
+
+    if [ $count -eq "$retries" ]; then
+        echo "Failed to install after $retries attempts. Exiting."
+        exit 1
+    fi
+}
+
+echo $'***** Installing s3fs *****\n'
+sudo rpm --import https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-8
+sudo dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+install_with_retry "sudo yum install -y s3fs-fuse" 3
+rpm -qa | grep epel-release
+rpm -qa | grep s3fs-fuse
+
+echo "======================Cloning HPC public repo====================="
+
+sudo yum install git -y
+mkdir /HPCaaS
+cd /HPCaaS
+git clone https://github.com/terraform-ibm-modules/terraform-ibm-hpc.git
+cd /HPCaaS/terraform-ibm-hpc/solutions/hpc
+
+echo "======================Installing terraform====================="
+git clone --depth=1 https://github.com/tfutils/tfenv.git ~/.tfenv
+echo "export PATH=$PATH:$HOME/.tfenv/bin" >> ~/.bashrc
+ln -s ~/.tfenv/bin/* /usr/local/bin
+tfenv install 1.5.7
```

Comment:
Do you have a plan for maintaining the versions installed by this script? For instance, we are already on terraform version 1.9.x

Reply:
I mean how do you keep the versions in this script updated?

Reply:
@daniel-butler-irl Good Point.. We can use the latest instead of pointing to the specific version.

We did a quick check and we came up with using the below:
tfenv install latest
tfenv use latest

Which shall install and use the latest version. For an example when I ran through it picked up 1.9.7


`[root@sagar-compute-rhel810 ~]# tfenv install latest 
Installing Terraform v1.9.7
Downloading release tarball from https://releases.hashicorp.com/terraform/1.9.7/terraform_1.9.7_linux_amd64.zip
############################################################################################################################################################################################################################################## 100.0%
Downloading SHA hash file from https://releases.hashicorp.com/terraform/1.9.7/terraform_1.9.7_SHA256SUMS
Not instructed to use Local PGP (/root/.tfenv/use-{gpgv,gnupg}) & No keybase install found, skipping OpenPGP signature verification
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LC_CTYPE = "UTF-8",
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("en_US.UTF-8").
Archive:  /tmp/tfenv_download.QFN1LC/terraform_1.9.7_linux_amd64.zip
  inflating: /root/.tfenv/versions/1.9.7/LICENSE.txt  
  inflating: /root/.tfenv/versions/1.9.7/terraform  
Installation of terraform v1.9.7 successful. To make this your default version, run 'tfenv use 1.9.7'
`

`[root@sagar-compute-rhel810 ~]# tfenv use latest
Switching default version to v1.9.7
Default version (when not overridden by .terraform-version or TFENV_TERRAFORM_VERSION) is now: 1.9.7
[root@sagar-compute-rhel810 ~]# terraform --version
Terraform v1.9.7
on linux_amd64
`

Hope this is as expected !